ajbleakley/register-app
# Feature - Install Docker

### What is the (feature|problem)?

Relates to: [AS A software engineer I WANT TO use a virtualisation tool SO THAT I can run the application consistently across all environments](https://github.com/users/ajbleakley/projects/1/views/1?pane=issue&itemId=65625366)

### What is the solution?

In my proposal I opted to use Docker and specifically a tool called [PHPDocker.io](https://phpdocker.io/) (recommended by [PHP The Right Way](https://phptherightway.com/#docker)).

[PHPDocker.io](https://phpdocker.io/) works great. I opted for a web server (nginx), a caching service (Memcached) and a relational database management system (MySQL) to persist data.

The only problem was the generated docker compose file was slightly outdated and hardcodes MySQL credentials into the public docker compose file. So I made a couple of small tweaks in [this commit](https://github.com/ajbleakley/register-app/commit/09379ce309e8923b4c0a593be3ef033424e494af).

### What areas of the app does it impact?

No impact to the app. Big impact for developers as it will allow them to quickly spin up the application container(s) in different environments.

### How to test?

Instructions:
1. Clone repository and checkout to this feature branch.
2. Run `composer install`
3. Copy and paste `.env.local.dist` to `.env.local` and overwrite default, unsecure database credentials **(optional but recommended step if running application locally)**.
4. Run `docker compose up -d` to build and serve the application environment.
5. Visit http://localhost:8000/api/ping in your web browser to verify the application is running.
6. You should also be able to connect to a database on port http://localhost:3306/ using database credentials specified in `.env.local` (or `.env` if not overwritten).
